### ISSUE TITLE
Line number background color?

### ISSUE THREAD

**Issue by OmegaVVeapon** [pygments/pygments#1773]
Hi,

I'm using the [HtmlFormatter](https://pygments.org/docs/formatters/?highlight=htmlformatter#HtmlFormatter) and after going through the available options, I could not find a way to set the line background color at all but which makes using line hyperlinks on dark themes (like `material`) pretty much impossible.

It's really hard to see this...
![image](https://user-images.githubusercontent.com/6247817/114448683-dcb7b900-9ba1-11eb-8685-c7420ea47dad.png)

After looking around some more I found issue https://github.com/pygments/pygments/issues/1206 and that's what I ended up doing...

```
sed -i 's/line_number_special_background_color.*/line_number_special_background_color = white/g' /usr/local/lib/python3.8/site-packages/pygments/styles/material.py
```

Which makes it much more readable:

![image](https://user-images.githubusercontent.com/6247817/114451994-ee02c480-9ba5-11eb-994f-7a0df3607883.png)

Is there another to set `line_number_special_background_color` that I don't know about that doesn't involve editing the code?

**Comment by Anteru** [comment]
That's a theme issue really. We have a way to set line number colors, and themes are supposed to set them correctly. If your links have colors which are no compatible I'm not sure how to fix that, because that's outside of our "reach". Can you try to inherit the link color in the CSS style? If that fixes it, maybe we should do that in the default CSS.

**Comment by heiyubai77** [comment]
Hi, I'm a student from the University of Adelaide. I'm working in group 10, team of 6 on contributing to an OSS project. We would like to tackle this issue, if it is still available.

**Comment by Anteru** [comment]
Yes it is. This might be a good chance to create a full "style checker" script -- we already have a check for color contrast which should be in theory able to capture this kind of problem, and it would be good if that was checked automatically because I presume this is not the only style suffering from it.

**Comment by heiyubai77** [comment]
Hello, we have implemented a few lines of code to satisfy the request. The changes to the original code is minor since the protection of code integrity. The idea is to make the line_number_background_color to be the complete opposite of the line_number_color automatically. However, we don't actually have a test structure to confirm whether this part of the code will be consistent with the original code. Could you please give us some advise on how we could test the output of the new code? And also, what else is needed to perfect this code. Here is the code part:
pygments>-style>-material

default_style = ""
    background_color = dark_teal
    highlight_color = '#2C3B41'
    line_number_color = '#37474F'
    line_number_color_hex = int(line_number_color[1:6], 16)
    line_number_color_reverse = int('FFFFFF', 16) - line_number_color_hex
    line_number_color_dark = '#' + str(line_number_color_reverse)
    line_number_background_color = line_number_color_dark
    line_number_special_color = '#607A86'
    line_number_special_background_color = line_number_color_dark

Thank you for your time.

**Comment by Jean-Abou-Samra** [comment]
I am not fully understanding what you want to achieve with this. Do you intend it to be part of the "material" style only? In that case, inverting the color automatically is a bit overkill -- just do `line_number_color_dark = "#NNNNNN" # opposite of dark_teal` (with a comment to explain your intent). Someone might want to tune that in the future so it won't be the exact opposite. Or do you intend this to be a default when themes don't set it?

**Comment by Woody98** [comment]
Thank you for your comment! What we've considered is that there might be some update to the code in the future. In case that someone is to change the coloring style of the line_number_color, they do not need to change the corresponding background color since the converting is automatic. However, we have seriously considered the fact that a complete opposite color might be a bit shiny for the eye. Therefore, we are trying to make the background into a color that is both opposing to the number color and mild for human eye.
